### PR TITLE
Correct percentage on parallel pull

### PR DESCRIPTION
**Pull request by abresas**
Newer versions of docker download multiple layers concurrently. As
events are processed by docker-progress, the current downloaded size
changes based on what layer is being downloaded now.

So, instead of having one variable for keeping track of currently
downloaded size, we have one entry per layer, and add them up
to get the aggregate downloaded size.

---

This change is [Reviewable](https://reviewable.io/reviews/resin-io/docker-progress/5)

### CODE REVIEW

**Comment by Page-** [comment]
Does this need a corresponding fix for push progress?

File: index.coffee [hunk 1]
```diff
@@ -157,29 +157,34 @@ exports.DockerProgress = class DockerProgress
...
 			completedSize = 0
+			layerDownloadedSize = {}
 			totalSize = _.sum(layerSizes)
 			return (evt) ->
 				{ status } = evt
 				if status == 'Downloading'
-					currentSize = evt.progressDetail.current
+					shortId = evt.id
+					longId = _.findKey(layerSizes, (v, id) -> _.startsWith(id, shortId))
+					if longId?
```
Comment (Page-) [inline index.coffee]: It would be nice to abstract the longId finding/checking/warning to a function rather than have it copy/pasted in two places here

**Comment by abresas** [comment]
@Page- it probably does, but I wanted to fix this part first as it's important for supervisor. I'll do push next.

File: index.coffee (hunk 1 above)
Comment (abresas) [inline index.coffee]: @Page- it was ugly to use one function for finding/checking/warning. I used a function for getting the long id and then improved error handling. please check, need to merge soon

**Comment by petrosagg** [comment]
SGTM

File: index.coffee [hunk 2]
```diff
@@ -96,6 +95,12 @@ onProgressHandler = (onProgressPromise, fallbackOnProgress) ->
 	# real onProgress function when the promise resolves
 	return (evt) -> onProgress(evt)
 
+getLongId = (shortId, layerIds) ->
+	longId = _.find(layerIds, (id) -> _.startsWith(id, shortId))
+	if not longId?
+		throw new Error("Progress error: Unknown layer #{shortId} downloaded by docker. Progress not correct.")
```
Comment (Page-) [inline index.coffee:101]: This can't throw the error - it'll crash the whole nodejs process

**Pull request merged**